src/actions/vstudio/vs2005_solution.lua

```diff
@@ -293,6 +294,32 @@
 
 
 --
+-- Write out the ExtensibilityGlobals block, which embeds some data for the
+-- Visual Studio PremakeExtension. In two seperate methods for override-ability.
+--
+	function sln2005.PremakeGlobals(wks)
+		-- we need to filter out the --file argument, since we already output
+		-- the script separately.
+		local args = {}
+		for _, arg in ipairs(_ARGV) do
+			if not (arg:startswith("--file") or arg:startswith("/file")) then
+				table.insert(args, arg);
+			end
+		end
+
+		_p(2, 'PremakeBinary = %s', _PREMAKE_COMMAND)
+		_p(2, 'PremakeScript = %s', p.workspace.getrelative(wks, _MAIN_SCRIPT))
+		_p(2, 'PremakeArguments = %s', table.concat(args, ' '))
+	end
+
+	function sln2005.ExtensibilityGlobals(wks)
+		_p(1, 'GlobalSection(ExtensibilityGlobals) = postSolution')
+		sln2005.PremakeGlobals(wks)
```

Comment:
It would probably be good to use a call array here instead, it allows modules to add more extensions and what not. :)